src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -265,7 +266,7 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
                 $id = $this->aliases[$id];
             }
             // Re-use shared service instance if it exists.
-            if (isset($this->services[$id])) {
+            if (isset($this->services[$id]) || array_key_exists($id, $this->services)) {
```

Comment:
But 2 calls instead of 1 in the worst case, and a code harder to read and maintain.